### PR TITLE
fix(openapi): failing recursion on api resources with "paths" key

### PR DESCRIPTION
**Pull request by pheadeaux**
| Q             | A
| ------------- | ---
| Branch?       | 2.6
| Tickets       | #4318
| License       | MIT

As described in issue #4318 the function recursiveClean does nest one more time and fails, if an ApiResource has a property named `paths`. Found no tests to reproduce or fail due to this fact.

### CODE REVIEW

**Comment by dunglas** [comment]
LGTM. Thanks for tracking and fixing this @pheadeaux. Could you add a new unit test to prevent future regressions please?

**Comment by pheadeaux** [comment]
@dunglas done. No need to add a new test, I just updated the existing one to cover the edge case. The Regressen happens when diving into the "Components>Schema" of an OpenApi Object.

File: src/OpenApi/Serializer/OpenApiNormalizer.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ private function recursiveClean($data): array
             }
 
             // Side effect of using getPaths(): Paths which itself contains the array
-            if ('paths' === $key) {
+            if ('paths' === $key && \is_array($value) && isset($data['paths']['paths'])) {
```
Comment (alanpoulain) [inline src/OpenApi/Serializer/OpenApiNormalizer.php:62]: Shouldn't the `is_array` use `$data` and not `$value`?

Comment (pheadeaux) [reply src/OpenApi/Serializer/OpenApiNormalizer.php:62]: @alanpoulain granted. But if you look up at [line 57](https://github.com/api-platform/core/pull/4319/files#diff-66a6198b9d63f63ba3a635fa8f23143870ed8c73842fba6b142b1c1a118b4e96R57), the if statement also directly checks the `$value` contents to be an array. This leads us to the question: check for `is_array($value)` at all or just remove it?

**Comment by pheadeaux** [comment]
@dunglas @alanpoulain Can somebody assist with the failing test? I do not see, how PHP-8 and Symfony 5.3 are differently affected by these changes than lets say PHP-7 and earlier versions of Symfony.

File: src/OpenApi/Serializer/OpenApiNormalizer.php (hunk 1 above)
Comment (alanpoulain) [reply src/OpenApi/Serializer/OpenApiNormalizer.php:62]: I think you can remove it.

**Comment by dunglas** [comment]
The test failure is unrelated, don't bother with it, I'm on it (#4323).

**Comment by alanpoulain** [comment]
We are fixing the PHP 8 / SF 5.3 tests, your PR has no impact on them.

**Comment by pheadeaux** [comment]
@alanpoulain @dunglas actually I have a better idea to solve the underlying problem. Please review #4325

**Comment by pheadeaux** [comment]
closed in favor of  #4325
